utils/components-testkit/src/main/scala/pl/touk/nussknacker/engine/util/test/ModelWithTestComponents.scala

```diff
@@ -0,0 +1,21 @@
+package pl.touk.nussknacker.engine.util.test
+
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigValueFactory.fromAnyRef
+import pl.touk.nussknacker.engine.ModelData
+import pl.touk.nussknacker.engine.api.component.ComponentDefinition
+import pl.touk.nussknacker.engine.api.process.EmptyProcessConfigCreator
+import pl.touk.nussknacker.engine.testing.LocalModelData
+import pl.touk.nussknacker.engine.testmode.{TestComponentsHolder, TestComponentsProvider}
+
+object ModelWithTestComponents {
+
+  //we add components by hand, which should be tested in scenario
+  //components are registered in special ComponentProvider, which is configured with appropriate testRunId
+  def prepareModelWithTestComponents(config: Config, components: List[ComponentDefinition]): ModelData = {
+    val testComponentHolder = TestComponentsHolder.registerTestComponents(components)
+    val configWithRunId = config.withValue(s"components.${TestComponentsProvider.name}.${TestComponentsProvider.testRunIdConfig}", fromAnyRef(testComponentHolder.runId.id))
```

Comment:
In the future would be nice to just created model date with components instead of wrapping them into component providers, exposing as a service and communicate with them via config...

Reply:
true, I added TODO in ProcessDefinitionExtractor